src/sentry/static/sentry/app/views/performance/landing/content.tsx

```diff
@@ -138,89 +137,83 @@ class LandingContent extends React.Component<Props, State> {
             </DropdownControl>
           </ProjectTypeDropdown>
         </SearchContainer>
-        {this.renderSelectedDisplay(currentLandingDisplay.field, summaryConditions)}
+        {this.renderSelectedDisplay(currentLandingDisplay.field)}
       </React.Fragment>
     );
   }
 
-  renderSelectedDisplay(display, summaryConditions) {
+  renderSelectedDisplay(display) {
     switch (display) {
       case LandingDisplayField.ALL:
-        return this.renderLandingAll(summaryConditions);
+        return this.renderLandingAll();
       case LandingDisplayField.FRONTEND_PAGELOAD:
-        return this.renderLandingFrontend(summaryConditions, true);
-      case LandingDisplayField.FRONTEND_NAVIGATION:
-        return this.renderLandingFrontend(summaryConditions, false);
+        return this.renderLandingFrontend(true);
+      case LandingDisplayField.FRONTEND_OTHER:
+        return this.renderLandingFrontend(false);
       case LandingDisplayField.BACKEND:
-        return this.renderLandingBackend(summaryConditions);
+        return this.renderLandingBackend();
       default:
         throw new Error(`Unknown display: ${display}`);
     }
   }
 
-  renderLandingFrontend = (summaryConditions, isPageload) => {
+  renderLandingFrontend = isPageload => {
     const {organization, location, projects, eventView, setError} = this.props;
 
-    const op = isPageload ? 'pageload' : 'navigation';
-
-    const frontendView = eventView.clone();
-    const conditions = tokenizeSearch(frontendView.query);
-    const newConditions = new QueryResults([]); // Add conditions just to include in summary as transaction.op you want to likely keep
-    conditions.setTagValues('transaction.op', [op]);
-    newConditions.setTagValues('transaction.op', [op]);
-
-    frontendView.query = stringifyQueryObject(conditions);
-    const frontendSummaryConditions = [
-      summaryConditions,
-      stringifyQueryObject(newConditions),
-    ].join(' ');
+    if (isPageload) {
+      eventView.addOverrideCondition('transaction.op', ['pageload']);
+    } else {
+      eventView.addOverrideCondition('!transaction.op', ['pageload']);
+    }
```

Comment:
Could this have been added in `generateFrontendPageloadPerformanceEventView` and `generateFrontendOtherPerformanceEventView` respectively instead of here?

Reply:
I'll take a look